src/polyhedralGravity/model/TreeNode.h

```diff
@@ -0,0 +1,35 @@
+#pragma once
+
+#include "KdDefinitions.h"
+
+/**
+* Abstract super class for nodes in the {@link KDTree}.
+*/
+class TreeNode {
+public:
+    virtual ~TreeNode() = default;
+    TreeNode(const TreeNode &) = delete;
```

Comment:
I did that in order to prevent copying objects unnecessarily during development. That way I get an error immediately and can resolve the error. Also it didn't make any sense to me why one should be able to copy TreeNodes which is why I deleted this alltogether.